texar/torch/modules/decoders/xlnet_decoder.py

```diff
@@ -281,10 +279,15 @@ def step(self,
         logits = F.linear(net_output, self.word_embed.weight, self.lm_bias)
         logits = logits[:, -1]
         sample_ids = helper.sample(time=time, outputs=logits)
-        (finished, next_inputs) = helper.next_inputs(
-            self.embed_tokens, time, logits, sample_ids)
         outputs = XLNetDecoderOutput(logits=logits, sample_id=sample_ids)
-        return outputs, memory, next_inputs, finished
+        return outputs, memory
+
+    def compute_next_input(self, helper: Helper, time: int,
```

Comment:
Is `compute_next_input` identical for all the derived classes? If it is, let's make it the default impl. in the base class.